modules/rte_mis_core/src/Helper/ConfigManager.php

```diff
@@ -0,0 +1,519 @@
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Theme Manager service.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected $themeManager;
+
+  /**
+   * Module Handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * Logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  private $logger;
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  private $fileSystem;
+
+  /**
+   * Constructs a new ConfigManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config storage object.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   Language Manager.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity Type Manager.
+   * @param \Drupal\Core\Theme\ThemeManagerInterface $theme_manager
+   *   Theme Manager.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   Module Handler.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   Logger Channel Factory.
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   *   The file system service.
+   */
+  public function __construct(
+    ConfigFactoryInterface $config_factory,
+    LanguageManagerInterface $language_manager,
+    EntityTypeManagerInterface $entity_type_manager,
+    ThemeManagerInterface $theme_manager,
+    ModuleHandlerInterface $module_handler,
+    LoggerChannelFactoryInterface $logger_factory,
+    FileSystemInterface $file_system,
+  ) {
+    $this->configFactory = $config_factory;
+    $this->languageManager = $language_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->themeManager = $theme_manager;
+    $this->moduleHandler = $module_handler;
+    $this->logger = $logger_factory->get('rte_mis_config');
+    $this->fileSystem = $file_system;
+  }
+
+  /**
+   * Update Config from code to active storage.
+   *
+   * @param array $configs
+   *   The name of configs to import.
+   * @param string $module_name
+   *   Name of the module, where files resides.
+   * @param string $path
+   *   Path where configs reside. Defaults to install.
+   * @param string $mode
+   *   Mode of update operation replace / add missing.
+   * @param array $options
+   *   Array of keys to replace when using MODE_REPLACE_KEY.
+   */
+  public function updateConfigs(array $configs, $module_name, $path = 'install', $mode = self::MODE_ADD_MISSING, array $options = []) {
+    if (empty($configs)) {
+      return;
+    }
+
+    if (!in_array($path, ['install', 'optional'])) {
+      $this->logger->error('If you are trying to apply the overrides, please resave (MODE_RESAVE) original config.');
+      throw new \InvalidArgumentException('Only original config should be updated.');
+    }
```

Comment:
Please update the logger message here with a proper one. Like only `install` `optional` configuration will be considered.